catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/utils/TableOpsUtils.java

```diff
@@ -18,20 +18,127 @@
     checkColumnAutoIncrement(fieldName, autoIncrement);
   }
 
+  public static List<SchemaChange> buildSchemaChanges(TableChange... tableChanges)
+      throws UnsupportedOperationException {
+    List<SchemaChange> schemaChanges = new ArrayList<>();
+    for (TableChange tableChange : tableChanges) {
+      schemaChanges.add(buildSchemaChange(tableChange));
+    }
+    return schemaChanges;
+  }
+
+  public static SchemaChange buildSchemaChange(TableChange tableChange)
+      throws UnsupportedOperationException {
+    if (tableChange instanceof ColumnChange) {
+      String[] fieldNames = ((ColumnChange) tableChange).fieldName();
+      checkNestedColumn(fieldNames);
+
+      if (tableChange instanceof AddColumn) {
+        AddColumn addColumn = (AddColumn) tableChange;
+        String fieldName = fieldName(addColumn);
+        checkColumnCapability(fieldName, addColumn.getDefaultValue(), addColumn.isAutoIncrement());
+        return addColumn(
+            fieldName,
+            toPaimonType(addColumn.getDataType()).copy(addColumn.isNullable()),
+            addColumn.getComment(),
+            move(fieldName, addColumn.getPosition()));
+      } else if (tableChange instanceof DeleteColumn) {
+        return dropColumn(fieldName((DeleteColumn) tableChange));
+      } else if (tableChange instanceof RenameColumn) {
+        RenameColumn renameColumn = ((RenameColumn) tableChange);
+        return renameColumn(fieldName(renameColumn), renameColumn.getNewName());
+      } else if (tableChange instanceof UpdateColumnComment) {
+        UpdateColumnComment updateColumnComment = (UpdateColumnComment) tableChange;
+        return updateColumnComment(
+            fieldName(updateColumnComment), updateColumnComment.getNewComment());
+      } else if (tableChange instanceof UpdateColumnNullability) {
+        UpdateColumnNullability updateColumnNullability = (UpdateColumnNullability) tableChange;
+        return updateColumnNullability(
+            fieldName(updateColumnNullability), updateColumnNullability.nullable());
+      } else if (tableChange instanceof UpdateColumnPosition) {
+        UpdateColumnPosition updateColumnPosition = (UpdateColumnPosition) tableChange;
+        return updateColumnPosition(
+            move(fieldName(updateColumnPosition), updateColumnPosition.getPosition()));
+      } else if (tableChange instanceof UpdateColumnType) {
+        UpdateColumnType updateColumnType = (UpdateColumnType) tableChange;
+        return updateColumnType(
+            fieldName(updateColumnType), toPaimonType(updateColumnType.getNewDataType()));
+      }
+    } else if (tableChange instanceof UpdateComment) {
+      return updateComment(((UpdateComment) tableChange).getNewComment());
+    } else if (tableChange instanceof SetProperty) {
+      SetProperty setProperty = ((SetProperty) tableChange);
+      String setPropertyKey = setProperty.getProperty();
+      checkReservedProperties(setPropertyKey);
```

Comment:
reserved properties are not allowed to alter, no need to check again

Reply:
removed